Okta.Xamarin/Okta.Xamarin/OktaStateManager.cs

```diff
@@ -195,17 +197,36 @@ public async Task<string> RenewAsync()
         /// Calls the OpenID Connect UserInfo endpoint with the stored access token to return user claim information.  This is an async method and should be awaited.
         /// </summary>
         /// <returns>A Task with a<see cref="System.Security.Claims.ClaimsPrincipal"/> representing the current user</returns>
-        public async Task<System.Security.Claims.ClaimsPrincipal> GetUserAsync()
+        public async Task<ClaimsPrincipal> GetUserAsync(string authorizationServerId = "default")
         {
-            throw new NotImplementedException();
+            string response = await PerformAuthorizationServerRequestAsync(HttpMethod.Get, "/userinfo", new Dictionary<string, string>
+            {
+                {"Authorization", $"Bearer {AccessToken}" }
+            }, authorizationServerId);
+
+            UserInfo userInfo = JsonConvert.DeserializeObject<UserInfo>(response);
+
+            ClaimsIdentity claimsIdentity = new ClaimsIdentity(userInfo.ToClaims(), "Okta");
+            ClaimsPrincipal claimsPrincipal = new ClaimsPrincipal(claimsIdentity);
+            return claimsPrincipal;
+
         }
 
         /// <summary>
-        /// Removes the local authentication state by removing cached tokens in the keychain. A full sign out should consist of calling <see cref="OidcClient.SignOutOfOktaAsync"/>, then <see cref="RevokeAsync"/>, and then <see cref="Clear"/>.
+        /// Removes the local authentication state. A full sign out should consist of calling <see cref="OidcClient.SignOutOfOktaAsync"/>, then <see cref="RevokeAsync"/>, and then <see cref="Clear"/>.
         /// </summary>
         public void Clear()
         {
-            throw new NotImplementedException();
+            AccessToken = string.Empty;
+            IdToken = string.Empty;
+            RefreshToken = string.Empty;
+            Scope = string.Empty;
+            Expires = null;
+        }
+
+        protected async Task<string> PerformRequestAsync(HttpMethod httpMethod, string path, Dictionary<string, string> headers)
```

Comment:
I suggest moving requests logic to the client, so you have all this logic in one place and it's easier to test. It feels like the state manager should only provide minimal functionality to manage your tokens. Thoughts?

Reply:
Another situation where I felt similarly at first; this implementation aligns more closely with how the iOS Sdk implements things.

Reply:
What about Android? Is Android using this pattern as well?